src/bun.js/node/node_fs.zig

```diff
@@ -5094,6 +5084,38 @@ pub const NodeFS = struct {
         };
     }
 
+    /// Calculate Base64 encoded string from an ArrayList of bytes.
+    /// Conversion happens in place, in the same input array, with an hardcoded chunk size of 16383 bytes.
+    /// Memory is allocated as needed and not at once time, preventing double allocations for input and output.
+    /// Returns an owned slice of the converted input.
+    fn base64Conversion(buf: *std.ArrayList(u8)) anyerror![]u8 {
+        // Chunk size should be a multiple of 3 bytes to be able to split base64 calculation
+        const chunk_size = 16383;
+        const encoded_chunk_size = comptime Base64.encodeLenFromSize(chunk_size);
+        var offset: usize = 0;
+
+        // Declare buffer to store encoded Base64 Chunk over every chunked input
+        const temp_encoding_buffer = bun.default_allocator.alloc(u8, encoded_chunk_size) catch |err| return err;
```

Comment:
i think we can get away with a single stack allocated buffer to reduce allocations

```
var encode_buf: [chunk_size]u8 = undefined;
```